src/main/java/com/devonfw/devcon/modules/oasp4j/Oasp4j.java

```diff
@@ -92,10 +92,10 @@ public Oasp4j() {
   public void create(String serverpath, String servername, String packagename, String groupid, String version,
       String dbtype) throws Exception {
 
-    Optional<String> oaspTemplateVersion_op = Downloader.getDevconConfigProperty(Constants.OASP_TEMPLATE_VERSION); // Optional.of("2.3.0");
-    String oaspTemplateVersion = oaspTemplateVersion_op.isPresent() ? oaspTemplateVersion_op.get()
-        : Constants.OASP_TEMPLATE_LAST_STABLE_VERSION;
-    if (!oaspTemplateVersion_op.isPresent())
+    serverpath = serverpath.isEmpty() ? getContextPathInfo().getCurrentWorkingDirectory().toString() : serverpath;
+
+    String oaspTemplateVersion = getTemplateVersion(serverpath);
+    if (oaspTemplateVersion.isEmpty())
```

Comment:
Should be not empty.
Seems, that this is not tested! Please add tests.

Reply:
I guess it could be empty in rare cases like file missing or file corruption. 
I will add some tests for the "getTemplateVersion" method  creating a temporary folder and a dummy "version.json" file because the "DevonServerPath" location is unknown (AFAIK).

Reply:
I just meant, looking at the diff, it seems, that you invalidated previous business logic. Beforehand, the if clause was checkinf if the version is >not empty<.

Reply:
OK! I will try to keep previous "if" condition by using an Optional<String> datatype